Wikipedia/Code/ArticleFullWidthImageCollectionViewCell.swift

```diff
@@ -92,7 +92,7 @@ open class ArticleFullWidthImageCollectionViewCell: ArticleCollectionViewCell {
 
         if !isSaveButtonHidden {
             origin.y += spacing - 1
-            let saveButtonFrame = saveButton.wmf_preferredFrame(at: origin, maximumWidth: widthMinusMargins, horizontalAlignment: isDeviceRTL ? .right : .left, apply: apply)
+            let saveButtonFrame = saveButton.wmf_preferredFrame(at: origin, maximumWidth: size.width, horizontalAlignment: isDeviceRTL ? .right : .left, apply: apply)
```

Comment:
This should stay `widthMinusMargins`, the bug is in `UIView+SemanticContent.swift`:

```
extension AlignedImageButton {
    override public func wmf_preferredFrame(at point: CGPoint, maximumSize: CGSize = NoIntrinsicSize, minimumSize: CGSize = NoIntrinsicSize, horizontalAlignment: HorizontalAlignment = .center, verticalAlignment: VerticalAlignment = .center, apply: Bool = false) -> CGRect  {
        let adjustedPoint = CGPoint(x: point.x - leftPadding, y: point.y - verticalPadding)
        var adjustedSize = maximumSize
        if adjustedSize.width != UIViewNoIntrinsicMetric {
            adjustedSize.width = adjustedSize.width + leftPadding + rightPadding
        }
        return super.wmf_preferredFrame(at: adjustedPoint, maximumSize: maximumSize, minimumSize: minimumSize, horizontalAlignment: horizontalAlignment, verticalAlignment: verticalAlignment, apply: apply)
    }
}
```

Reply:
oh cool thx!

Reply:
It pushes the frame to account for the extra button space, it's hardcoded for .left and .top alignment, it needs to be updated to respect the other alignments

Reply:
turns out we weren't using the `adjustedSize` in that override

Reply:
Looks better now: 

| Before  | After |
| ------------- | ------------- |
| <img width="431" alt="screen shot 2018-07-25 at 1 34 27 pm" src="https://user-images.githubusercontent.com/3143487/43226047-936c382a-900f-11e8-9b1a-c2a97e19aaa2.png"> | <img width="431" alt="screen shot 2018-07-25 at 1 33 31 pm" src="https://user-images.githubusercontent.com/3143487/43226041-8ff99836-900f-11e8-9192-b5ef30059dfe.png"> |